src/types.rs

```diff
@@ -303,34 +325,44 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
+      original_name : NameType::new(digest.0),
```

Comment:
Are we ever going to reuse the `original_name`? If not, we can just store the `is_relocated` information in a `bool`. Ids are going to be copied over the network, so this would be a wasted 64Bytes.